includes/init_includes/init_customer_auth.php

```diff
@@ -79,27 +79,34 @@
  * 2 = customer authorization pending can browse but no prices
  */
 switch (true) {
+  /**
+   * bypass redirects for these scripts, to processing regardless of store mode or cust auth mode
+   */
+  case (preg_match('/ipn_main_handler\.php$/', $_SERVER['SCRIPT_NAME'])):

```

Comment:
Do we want to match '/some-random-folder-inside-zencart-installation-folder/ipn_main_handler.php'? Or should it be a more specific?

EX: '~^' . ($request_type == 'SSL' ? DIR_WS_HTTPS_CATALOG : DIR_WS_CATALOG) . 'ipn_main_handler\.php$~'?